packages/webcomponents/src/api/Api.ts

```diff
@@ -2,12 +2,12 @@ import { v4 as uuidv4 } from 'uuid';
 import { PagingInfo } from './Pagination';
 
 export interface IApiResponse<T> {
-  data: T;
+  data?: T;
   error?: IErrorObject | IServerError;
   page_info?: PagingInfo;
   errors?: string[];
-  id: number;
-  type: string;
+  id?: number;
+  type?: string;
 }
```

Comment:
This change seems fine, but just wanted to check and ask if there is anything in this PR that requires this change? 

I couldn't see anything, and if not, thats OK. Just curious really. 

It makes sense for me to have these as optional properties though.

Reply:
I had to add this because in the tests files, whenever I had to return an error object (the same we get from API), Typescript was complaining that the return type was missing id, type and the other properties. So these properties are really optional.